### PR TITLE
Avatar grunts breaks speech phoneme recognition

### PR DESCRIPTION
**Pull request by tcm390**
The issue is caused by that both `audioWorker` and `microphoneWorker` are setting the value of `this.volume`

related:
https://github.com/webaverse/app/issues/3645

Result:

https://user-images.githubusercontent.com/60634884/185251426-803f8f20-16d5-4c41-9b26-8f746012d6e0.mp4

### CODE REVIEW

**Review by avaer — CHANGES_REQUESTED** [review]
If volume goes from 1 -> 0 in one frame won't this cause it to get stuck? That's probably an even worse effect than the bug it is fixing.

How about choose which one to emit when it's enabled, not just by chance when it's zero?

**Comment by tcm390** [comment]
[disable audio worker after grunt end](https://github.com/webaverse/app/pull/3651/commits/2d7e702001a61e38479d372ce4a6d6b00aae6787)

> If volume goes from 1 -> 0 in one frame won't this cause it to get stuck? That's probably an even worse effect than the bug it is fixing.
> 
> How about choose which one to emit when it's enabled, not just by chance when it's zero?

**Comment by avaer** [comment]
> Can we disable audio worker after the grunt effect is completed?
No I don't think that's right, creating/destroying workers is a pretty heavy operation to run on every sound effect...

**Comment by tcm390** [comment]
> > Can we disable audio worker after the grunt effect is completed?
> > No I don't think that's right, creating/destroying workers is a pretty heavy operation to run on every sound effect...

[disable audioWorker to set the volume when isGrunting is false](https://github.com/webaverse/app/pull/3651/commits/3413f9b23fd4fefccb5d280f6bd21b8282ec0517)

Got it. Is it okay to use `isGrunting` in avatar.js to decide whether `audioWorker` could set volume or not.

**Comment by avaer** [comment]
> Got it. Is it okay to use isGrunting in avatar.js to decide whether audioWorker is allowed to set volume or not?

Probably not, because I think multiple grunts can happen at a time. At the very least I think this would need to be a stack number, not a boolean.

And if this is related to `manuallySetMouth`, shouldn't both be in the same class? Currently it's spread around so the code is confusing to read.

**Comment by tcm390** [comment]
> And if this is related to `manuallySetMouth`, shouldn't both be in the same class? Currently it's spread around so the code is confusing to read.

[disable audioworker set volume after grunting](https://github.com/webaverse/app/pull/3651/commits/e495dbc63b728fe81e2ad3401b7378e1edd72d70)

Got it. Renamed `characterBehavior.manuallySetMouth` to `characterBehavior.disableAudioWorkerSetVolume`. And used `characterBehavior.disableAudioWorkerSetVolume` to disable audioWorker after emitting grunt.

File: avatars/avatars.js [hunk 1]
```diff
@@ -993,7 +993,7 @@ class Avatar {
     this.lastNeedsHeadTarget = false;
     this.lastHeadTargetTime = -Infinity;
 
-    this.manuallySetMouth=false;
+    this.disableAudioWorkerSetVolume=false;
```
Comment (avaer) [inline avatars/avatars.js:996]: It is not good practice to have double negatives in variable names. The variable name should be the positive version, `enable`.

Comment (avaer) [inline avatars/avatars.js:996]: The default seems to be `true` when the grunt ends... so why is this false. Probably a consequence of choosing a poor variable name.

Comment (tcm390) [reply avatars/avatars.js]: Got it. [set disableAudioWorkerSetVolume initial value to true](https://github.com/webaverse/app/pull/3651/commits/cb447d56a76c188fa794dded6a04371269565571)

**Comment by avaer** [comment]
It looks like this is still using the `disable` name: https://github.com/webaverse/app/pull/3651/files#diff-fcb2093ac11d9b04a179d3dedbef5c0aae4d14b7f71db90e483becca4170b7ccR2041

**Comment by tcm390** [comment]
> It looks like this is still using the `disable` name: https://github.com/webaverse/app/pull/3651/files#diff-fcb2093ac11d9b04a179d3dedbef5c0aae4d14b7f71db90e483becca4170b7ccR2041

Sorry Changed it.
[change disableAudioWorkerSetVolume to enableAudioWorkerSetVolume](https://github.com/webaverse/app/pull/3651/commits/82d734e63e5371d383d5c3a6ea61399211339287)

**Comment by rondoor124** [comment]
I have QA it and seems like work fine.

**Pull request closed without merging**